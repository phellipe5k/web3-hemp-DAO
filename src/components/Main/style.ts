import styled from 'styled-components';

export const Container = styled.main`
  background-color: #345df6;
  color: #fff;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  padding: 3%;
  text-align: center;
  justify-content: center;
  align-items: center;
`;

export const Logo = styled.img`
  width: 25rem;
`;

export const Title = styled.h1`
  font-size: 4rem;
  padding: 2% 0;
`;

export const Description = styled.p`
  font-size: 2rem;
  padding: 2% 0;
  width: 60%;
`;

export const Illustration = styled.img`
  width: min(35rem, 100%);
  background-color: white;
  padding: 0.5%;
  border-radius: 15px;
`;
