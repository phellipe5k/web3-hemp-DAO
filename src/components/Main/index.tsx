import * as S from './style';

// Icons
import { NextDotJs as NextJsIcon } from '@styled-icons/simple-icons';

type Props = {
  title?: string;
  description?: string;
  illustrationSrc?: string;
};

const Main = ({ title, description, illustrationSrc }: Props) => (
  <S.Container>
    <NextJsIcon color="white" />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
    <S.Illustration src={illustrationSrc} />
  </S.Container>
);

export default Main;
