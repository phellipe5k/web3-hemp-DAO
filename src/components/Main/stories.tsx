import { Meta, Story } from '@storybook/react';
import Main from '.';

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'React Avançado',
    description: 'Crie aplicações reais com NextJS, Strapi, GraphQL e mais!',
    illustrationSrc:
      'https://d33wubrfki0l68.cloudfront.net/cf9dfe86983925690340feb07037ee260f872e34/74dff/img/hero-illustration.svg'
  }
} as Meta;

export const Basic: Story = (args) => <Main {...args} />;
