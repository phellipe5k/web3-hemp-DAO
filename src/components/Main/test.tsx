import { render, screen } from '@testing-library/react';

import Main from '.';

describe('<Main/>', () => {
  const { container } = render(
    <Main
      title="React Avançado"
      description="Crie aplicações reais com NextJS, Strapi, GraphQL e mais!"
      illustrationSrc="https://d33wubrfki0l68.cloudfront.net/cf9dfe86983925690340feb07037ee260f872e34/74dff/img/hero-illustration.svg"
    />
  );

  it('should everything be centered on screen', () => {
    expect(container.firstChild).toHaveStyle({ 'align-items': 'center' });
  });

  it('should snapshot', () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
