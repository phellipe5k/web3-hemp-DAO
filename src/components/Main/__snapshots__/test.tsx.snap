// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> should snapshot 1`] = `null`;
